src/predicates.ts

```diff
@@ -798,3 +799,132 @@ export class NotBoolString implements Predicate<any> {
     return { matched: true, paths: [this.path] };
   }
 }
+
+/**
+ * Invalid URL matches when a url is not valid or does not use
+ * `http` or `https` for the scheme
+ */
+export class InvalidURL implements Predicate<any> {
+  public static fromJson(self: any): InvalidURL {
+    return new InvalidURL(self.path);
+  }
+
+  constructor(
+      private readonly path: string,
+  ) {
+  }
+
+  public test(root): RuleMatchedAt {
+    const val = _.get(root, this.path);
+    const parsed = urlParse(val);
+    if (parsed.protocol === "http:" || parsed.protocol === "https:") {
```

Comment:
Do you need to check hostname like libyaml too?  There are some good bad url examples here https://replicated.atlassian.net/browse/VENAPI-180